module_utils/network_lsr/argument_validator.py

```diff
@@ -531,6 +533,13 @@ def _validate_post(self, value, name, result):
                 raise ValidationError(
                     name, "'dhcp4_send_hostname' is only valid if 'dhcp4' is enabled"
                 )
+        if result["ipv6_disabled"] is True:
+            result["auto6"] = None
+        if result["auto6"] is not None:
```

Comment:
at this point, `result["auto6"]` cannot be None.

Reply:
if `result["auto6"]` is not None. then the error `"'auto6' and 'ipv6_disabled' are mutually exclusive"`  will be invoked everytime based on the following testing code:
```yaml
        network_connections:
          - name: eth0
            type: ethernet
            ip:
              ipv6_disabled: true
```

Reply:
above there is the code
```
        if result["auto6"] is None:  
            result["auto6"] = not any(
                [a for a in result["address"] if a["family"] == socket.AF_INET6]
            )
```
so here the condition `if result["auto6"] is not None:` is NEVER satisfied.

Reply:
> so here the condition if result["auto6"] is not None: is NEVER satisfied.

Wrong of course. "is ALWAYS satisfied".

Reply:
I think the validation code is good now.